Kubeflow community Distribution 26.03 and 26.03.01 blog (#200)
kubeflow/blog: _posts/2026-04-11-kubeflow-26.03-release.md

---

> title: "Kubeflow AI Reference Platform 26.03 Release Announcement"

Reviewer: Let's call it Kubeflow Community Distribution 26.03

> title: "Kubeflow AI Reference Platform 26.03 Release Announcement"

Reviewer 2: ```suggestion
title: "Kubeflow Community Distribution 26.03.1 Release Announcement"
```

This release improves security and multi-tenancy through stronger Pod Security Standards enforcement and expanded network policy coverage across key system components. It also enhances operational efficiency with simplified installation, CI-tested upgrade paths, optimized Pod Disruption Budgets, and removal of legacy synchronization steps for faster and more reliable deployments.

In addition, it strengthens reliability through expanded KServe testing for both authenticated and raw deployments, improves CI coverage across Kubernetes environments, and streamlines installation by automating tooling selection for more consistent and reproducible upgrades.

This release adopts a calendar-based versioning model (Year.Month.Patch), with two primary releases annually and optional patches. Community support is best-effort for approximately six months, with additional commercial support options available  Regular upgrades are recommended to take advantage of continuous security and performance enhancements.

> Refernce: [Release Handbook](https://github.com/kubeflow/community/blob/master/releases/handbook.md) and [Kubeflow AI Reference Platform Roadmap](https://github.com/kubeflow/kubeflow/blob/master/ROADMAP.md)

Reviewer 3: ```suggestion
Refernce: [Release Handbook](https://github.com/kubeflow/manifests/blob/master/releases/kubeflow-ai-reference-platform-release-handbook.md) and [Kubeflow AI Reference Platform Roadmap](https://github.com/kubeflow/kubeflow/blob/master/ROADMAP.md)
```

> Refernce: [Release Handbook](https://github.com/kubeflow/manifests/blob/master/releases/kubeflow-ai-reference-platform-release-handbook.md) and [Kubeflow AI Reference Platform Roadmap](https://github.com/kubeflow/kubeflow/blob/master/ROADMAP.md)

Reviewer 2: ```suggestion
The release versioning is now calendar-based (Year.Month.Patch). Around two base releases are [planned](https://github.com/kubeflow/manifests/blob/master/releases/kubeflow-ai-reference-platform-release-handbook.md) per year with optional patch releases. The [best-effort only community support](https://www.kubeflow.org/docs/started/support/#support-from-the-kubeflow-community) is roughly 6 months and there is [commercial support](https://www.kubeflow.org/docs/started/support/#support-from-commercial-providers-in-the-kubeflow-ecosystem) available from multiple vendors. Please update regularly as explained in our [upgrading and extending section](https://github.com/kubeflow/manifests?tab=readme-ov-file#upgrading-and-extending) to benefit also from security and performance improvements.
```

> ## Highlight features 

Reviewer 2: ```suggestion
## Highlight features 26.03
```

* Kubernetes 1.34+
* Kubeflow Pipelines 2.16.0, Spark operator 2.5.0 Model registry v0.3.5, Kserve Web Application v0.16.1
* Compatibility of Kubeflow Pipelines v1 and v2 with PSS restricted
* Extended Kserve test with authentication and authorization from inside and outside the cluster as well as non-knative / raw deployments 
* Simplified installation and automatic installation of the right Kustomize and Kubectl versions
* Installation steps tested and based on our CI, easier in-place updates (optimized PDBs)
* Cleanup of all synchronization steps for faster releases / updates of dependencies
* Knative 1.20, cert-manager 1.19.4, Oauth2-proxy v7.14.3, Dex 2.45.0
* Fix networkpolicies for cert-manager, knative-serving, istio-system, dex, oauth2-proxy

> ## Kubeflow 26.03.1 release features

Reviewer 2: ```suggestion
## Highlight features 26.03.1
```

* Kind 0.31+ and Kubernetes 1.35 CI tests
* Kubeflow Pipelines 2.16.1, Spark operator 2.5.0 Model registry v0.3.9, Kserve Web Application v0.18.0
* Trainer 2.2.0
* Kserve 0.18.0
* Kubeflow/notebooks v1 update and workspaces (v2) beta
* Kubeflow/dashboard v2
* Enable model registry and catalog
* Knative 1.20, cert-manager 1.19.4, Oauth2-proxy v7.14.3, Dex 2.45.0
* Istio 1.30.0 with hostUsers: false support
* CI, PSS restricted and network policies for the optional knative-eventing

> * Restructure Upgrading section with version-specific upgrade notes

Reviewer 2: ```suggestion
- Kind 0.32+ and Kubernetes 1.36 CI tests
- Kubeflow Pipelines 2.16.1, Spark operator 2.5.0 Model registry v0.3.9, Kserve Web Application v0.18.0
- Trainer 2.2.0
- Kserve 0.18.0
- Kubeflow/notebooks v1 update and workspaces (v2) beta
- Kubeflow/dashboard v2
- Enable model registry and catalog
- Knative 1.22.0, cert-manager 1.20.2, Oauth2-proxy v7.15.2, Dex 2.45.1
- Istio 1.30.1 with hostUsers: false support
- CI, PSS restricted and network policies for the optional knative-eventing
- Restructure Upgrading section with version-specific upgrade notes
```

| [1.10.0](https://github.com/kubeflow/kubeflow/releases/tag/v1.10.0) | [1.10.0](https://github.com/kubeflow/kubeflow/releases/tag/v1.10.0) | [2.16.0](https://github.com/kubeflow/pipelines/releases/tag/2.16.0) | [0.19.0](https://github.com/kubeflow/katib/releases/tag/v0.19.0) | [2.1.0](https://github.com/kubeflow/trainer/releases/tag/v2.1.0) | [0.16.0](https://github.com/kserve/kserve/releases/tag/v0.16.0) | [0.3.7](https://github.com/kubeflow/model-registry/releases/tag/v0.3.7) | [2.5.0](https://github.com/kubeflow/spark-operator/releases/tag/v2.5.0) | [0.3.0](https://github.com/kubeflow/sdk/releases/tag/0.3.0) |

| Kubernetes |  Kind  | Kustomize | Cert Manager | Knative | Istio |  Dex | OAuth2-proxy |
| :--------: | :----: | :-------: | :----------: | :-----: | :---: | :--: | :----------: |

> |  [1.35+] (https://github.com/kubernetes/kubernetes/releases/tag/v1.35.3)  | [0.30.1](https://github.com/kubernetes-sigs/kind/releases/tag/v0.31.0) |  [5.8.1](https://github.com/kubernetes-sigs/kustomize/releases/tag/kustomize%2Fv5.8.1) |  [1.20.2](https://github.com/cert-manager/cert-manager/releases/tag/v1.20.2)   |   [1.21.0](https://knative.dev/blog/releases/announcing-knative-v1-21-release/) |  [1.29.1](https://github.com/istio/istio/releases/tag/1.29.1) | [2.45.1](https://github.com/dexidp/dex/releases/tag/v2.45.1) |  [7.15.1](https://github.com/oauth2-proxy/oauth2-proxy/releases/tag/v7.15.1)       |

Reviewer 3: ```suggestion
|  [1.35+](https://github.com/kubernetes/kubernetes/releases/tag/v1.35.3)  | [0.30.1](https://github.com/kubernetes-sigs/kind/releases/tag/v0.31.0) |  [5.8.1](https://github.com/kubernetes-sigs/kustomize/releases/tag/kustomize%2Fv5.8.1) |  [1.20.2](https://github.com/cert-manager/cert-manager/releases/tag/v1.20.2)   |   [1.21.0](https://knative.dev/blog/releases/announcing-knative-v1-21-release/) |  [1.29.1](https://github.com/istio/istio/releases/tag/1.29.1) | [2.45.1](https://github.com/dexidp/dex/releases/tag/v2.45.1) |  [7.15.1](https://github.com/oauth2-proxy/oauth2-proxy/releases/tag/v7.15.1)       |
```

> |  [1.35+](https://github.com/kubernetes/kubernetes/releases/tag/v1.35.3)  | [0.30.1](https://github.com/kubernetes-sigs/kind/releases/tag/v0.31.0) |  [5.8.1](https://github.com/kubernetes-sigs/kustomize/releases/tag/kustomize%2Fv5.8.1) |  [1.20.2](https://github.com/cert-manager/cert-manager/releases/tag/v1.20.2)   |   [1.21.0](https://knative.dev/blog/releases/announcing-knative-v1-21-release/) |  [1.29.1](https://github.com/istio/istio/releases/tag/1.29.1) | [2.45.1](https://github.com/dexidp/dex/releases/tag/v2.45.1) |  [7.15.1](https://github.com/oauth2-proxy/oauth2-proxy/releases/tag/v7.15.1)       |

Reviewer 2: the numbers here are not correct anymore for 26.03.1

Author: I will validate the version and compare it with [latest release](https://github.com/kubeflow/community-distribution/releases/tag/26.03.1)

> |  [1.35+](https://github.com/kubernetes/kubernetes/releases/tag/v1.35.3)  | [0.30.1](https://github.com/kubernetes-sigs/kind/releases/tag/v0.31.0) |  [5.8.1](https://github.com/kubernetes-sigs/kustomize/releases/tag/kustomize%2Fv5.8.1) |  [1.20.2](https://github.com/cert-manager/cert-manager/releases/tag/v1.20.2)   |   [1.21.0](https://knative.dev/blog/releases/announcing-knative-v1-21-release/) |  [1.29.1](https://github.com/istio/istio/releases/tag/1.29.1) | [2.45.1](https://github.com/dexidp/dex/releases/tag/v2.45.1) |  [7.15.1](https://github.com/oauth2-proxy/oauth2-proxy/releases/tag/v7.15.1)       |

Reviewer 2: ```suggestion
```

Author: @juliusvonkohout Are we not going to have this section in the blog or do I need to add with corrected version?

### Security:

> * **Identity, AuthN/AuthZ & Access Control

Reviewer 2: ```suggestion
* **Identity, auththenication, authorization & Access Control
```

    * Dex OIDC config fixes ([#3299](https://github.com/kubeflow/manifests/pull/3299)) 
    * oauth2-proxy upgrade ([#3366](https://github.com/kubeflow/manifests/pull/3366)) 
    * Block impersonating service accounts ([#3372](https://github.com/kubeflow/manifests/pull/3372))
    
* **Pod / Cluster Security Hardening
    * Pod Security Standards set to “baseline” ([#3414](https://github.com/kubeflow/manifests/pull/3414))
    * Knative eventing security overlay ([#3419](https://github.com/kubeflow/manifests/pull/3419))
    * NetworkPolicies variable namespace fix ([#3342](https://github.com/kubeflow/manifests/pull/3342))

* **Vulnerability Scanning / CI Security

>     * Remove Trivy scanning from manifests CI ([#3424](https://github.com/kubeflow/manifests/pull/3424))

Reviewer 2: That is not something we should advertise

* **TLS/Certificate & Infrastructure Security
    * cert-manager updates ([#3358](https://github.com/kubeflow/manifests/pull/3358), [#3370](https://github.com/kubeflow/manifests/pull/3370), [#3410](https://github.com/kubeflow/manifests/pull/3410))
    * Istio upgrades ([#3359](https://github.com/kubeflow/manifests/pull/3359), [#3456](https://github.com/kubeflow/manifests/pull/3456), [#3467](https://github.com/kubeflow/manifests/pull/3467))

* **Runtime / Platform Improvements
    * Webhook readiness hardening ([#3367](https://github.com/kubeflow/manifests/pull/3367))

>     * KServe / model registry / pipelines upgrades

Reviewer 2: ```suggestion
```

## [Pipelines ](https://github.com/kubeflow/pipelines)

## [Model Registry](https://github.com/kubeflow/hub)

## [Training Operator (Trainer) ](https://github.com/kubeflow/trainer)

## [Katib](https://github.com/kubeflow/katib)

## [Spark Operator](https://github.com/kubeflow/spark-operator)

## [KServe](https://github.com/kserve/kserve)

## [Kubeflow SDK](https://github.com/kubeflow/sdk)

> ## [Dashboard](https://github.com/kubeflow/dashboard)

Author: @thesuperzapper I’ve split this into two sections. I noticed that the Dashboard and Notebooks components have been moved to a new repository in 26.03.1. They were part of a https://github.com/kubeflow/kubeflow previously? 

What are some of the things I can add to this section ?

Reviewer 4: Hey Danisha, did you already see the release presentation? 🤗 That includes all the things that I presented as part of the release video, if you have additional questions feel free to reach out!

## [Notebooks](https://github.com/kubeflow/notebooks)

## How to get started with 26.03

Visit the Kubeflow AI Reference Platform 26.03 [release page](https://github.com/kubeflow/manifests/releases) or head over to the Getting Started and Support pages.

## Join the Community

We would like to thank everyone who contributed to Kubeflow 26.03, and especially Tarek Abouzeid for his work as the v26.03 Release Manager. We also extend our thanks to the entire release team and the working group leads, who continuously and generously dedicate their time and expertise to Kubeflow.

> Release team members : Tarek Abouzeid, Anya Kramar, Andy Stoneberg, Humair Khan, Matteo Mortari, Adysen Rothman, Jon Burdo, Milos Grubjesic, Vraj Bhatt, Dhanisha Phadate, Alok Dangre

Reviewer 5: ```suggestion
Release team members : Tarek Abouzeid, Anya Kramar, Andy Stoneberg, Alyssa Goins, Matteo Mortari, Adysen Rothman, Jon Burdo, Milos Grubjesic, Vraj Bhatt, Dhanisha Phadate, Alok Dangre
```

- Kubeflow Hub: Prepare for v1 release by ensuring stable API
- Knative 1.22.0, cert-manager 1.20.2, oauth2-proxy v7.15.2, Dex 2.45.1
- Istio 1.30.1 with hostUsers: false support
- CI, PSS restricted and network policies for the optional knative-eventing
- Restructure Upgrading section with version-specific upgrade notes

## Kubeflow Community Distribution:

* [Documentation updates](https://github.com/kubeflow/community-distribution/blob/master/README.md) that make it easier to install, extend and upgrade Kubeflow

### Kubeflow Community Distribution 26.03

|Notebooks|Dashboard|Pipelines|Katib|Trainer|KServe|Model Registry|Spark|SDK|
| :------------------------------------------------------: | :---------------------------------------: | :--------------------------------------------------------: | :---------------------------------------: | :--------------------------------------------------------: | :---------------------------------------: | :--------------------------------------------------------: | :---------------------------------------: | :--------------------------------------------------------: |

> | [1.10.0](https://github.com/kubeflow/notebooks/releases/tag/v1.11.0) | [1.10.0](https://github.com/kubeflow/kubeflow/releases/tag/v1.10.0) | [2.16.0](https://github.com/kubeflow/pipelines/releases/tag/2.16.0) | [0.19.0](https://github.com/kubeflow/katib/releases/tag/v0.19.0) | [2.1.0](https://github.com/kubeflow/trainer/releases/tag/v2.1.0) | [0.16.0](https://github.com/kserve/kserve/releases/tag/v0.16.0) | [0.3.7](https://github.com/kubeflow/model-registry/releases/tag/v0.3.7) | [2.5.0](https://github.com/kubeflow/spark-operator/releases/tag/v2.5.0) | [0.3.0](https://github.com/kubeflow/sdk/releases/tag/0.3.0) |

Author: As per slides, the Notebook version for the 26.03 release is v1.10. At that time, Notebooks was still part of the main Kubeflow release repository. I'll update the link to point to the Kubeflow v1.10.0 release:
https://github.com/kubeflow/kubeflow/releases/tag/v1.10.0

- [Linux Foundation zoom link](https://zoom-lfx.platform.linuxfoundation.org/meeting/91692004209?password=e48ce50b-b41d-459a-9f32-7313fa388380)
- Join #kubeflow-spark-on-kubernetes Slack channel

## [KServe](https://github.com/kserve/kserve)

- Standardized GenAI inference
  - Introduced the LLMInferenceService CRD to make LLM serving a first-class, opinionated platform primitive instead of raw vendor-specific configurations
- Multinode LLM workloads & Production grade LLM autoscaling
  - 0.18 laid groundwork for distributed inference across multiple nodes without forcing Ray, using KServe own scheduler/LWS integration
  - Added Workload Verification Authority (WVA) + KEDA/HPA integration so LLM endpoints scale based on real inference demand, not just CPU/memory
- OpenAI-compatible APIs expanded
  - Added /v1/responses route for OpenAI Responses API; standardized REST/gRPC dual-protocol routing for Standard mode
- Model cache matured
  - Namespace-scoped ModelCache support plus configurable download job scheduling, reducing cold-start latency for shared models

## [Dashboard](https://github.com/kubeflow/dashboard) and [Notebooks](https://github.com/kubeflow/notebooks)

> - **Manifest restructuring:** The Dashboard manifests have undergone breaking changes. If you're upgrading, be sure to review the migration guide included in the release notes and the community distribution documentation.

Reviewer 4: ```suggestion
- **Manifest restructuring:** The Dashboard manifests have undergone breaking changes. If you're upgrading, be sure to review the migration guide included in the [release notes](https://github.com/kubeflow/dashboard/releases/tag/v2.0.0) and the [community distribution documentation](https://github.com/kubeflow/community-distribution/#2603---next-release).
```

---
title: "Kubeflow Community Distribution 26.03.1 Release Announcement"

> layout: post

Reviewer 2: no it is not.

toc: false
comments: true
image: images/logo.png
hide: false
categories: [release]
permalink: /kubeflow-26.03-release/

> author: "Kubeflow 26.03 Release Team"

Reviewer 2: no

---

> The release versioning is now calendar-based (Year.Month.Patch). Around two base releases are [planned](https://github.com/kubeflow/manifests/blob/master/releases/kubeflow-ai-reference-platform-release-handbook.md) per year with optional patch releases. The [best-effort only community support](https://www.kubeflow.org/docs/started/support/#support-from-the-kubeflow-community) is roughly 6 months and there is [commercial support](https://www.kubeflow.org/docs/started/support/#support-from-commercial-providers-in-the-kubeflow-ecosystem) available from multiple vendors. Please update regularly as explained in our [upgrading and extending section](https://github.com/kubeflow/manifests?tab=readme-ov-file#upgrading-and-extending) to benefit also from security and performance improvements.

Reviewer 2: The links are wrong! it must be https://github.com/kubeflow/community-distribution/releases/tag/26.03

## Highlight features 26.03

* Kubernetes 1.34+
* Kubeflow Pipelines 2.16.0, Spark operator 2.5.0 Model registry v0.3.5, Kserve Web Application v0.16.1
* Compatibility of Kubeflow Pipelines v1 and v2 with PSS restricted
* Extended KServe tests with authentication and authorization from inside and outside the cluster as well as non-knative / raw deployments
* Simplified installation and automatic installation of the right Kustomize and Kubectl versions
* Installation steps tested and based on our CI, easier in-place updates (optimized PDBs)
* Cleanup of all synchronization steps for faster releases / updates of dependencies
* Knative 1.20, cert-manager 1.19.4, oauth2-proxy v7.14.3, Dex 2.45.0
* Fix network policies for cert-manager, knative-serving, istio-system, dex, oauth2-proxy

## Highlight features 26.03.1

- Kind 0.32+ and Kubernetes 1.36 CI tests
- Kubeflow Pipelines 2.16.1, Spark operator 2.5.0 Model registry v0.3.9, Kserve Web Application v0.18.0
- Trainer 2.2.0
- Kserve 0.18.0
- Kubeflow/notebooks v1 update and workspaces (v2) beta
- Kubeflow/dashboard v2

> - Enable model registry and catalog

Reviewer 6: Some Model Registry / Kubeflow hub items. Feel free to pare down/adjust if these are too verbose!
```suggestion
- Enable model registry, catalog, and UI in default Kubeflow installation
- Complete renaming of Model Registry to Kubeflow Hub, capturing the project's broader scope
- Kubeflow Hub: Prepare for v1 release by ensuring stable API
```

- CI, PSS restricted and network policies for the optional knative-eventing
- Restructure Upgrading section with version-specific upgrade notes

## [Pipelines](https://github.com/kubeflow/pipelines)

## [Model Registry](https://github.com/kubeflow/hub)

## [Training Operator (Trainer)](https://github.com/kubeflow/trainer)

## [Katib](https://github.com/kubeflow/katib)

## [Spark Operator](https://github.com/kubeflow/spark-operator)

## [KServe](https://github.com/kserve/kserve)

## [Kubeflow SDK](https://github.com/kubeflow/sdk)

## [Dashboard](https://github.com/kubeflow/dashboard)

## [Notebooks](https://github.com/kubeflow/notebooks)

## How to get started with 26.03

Visit the community distribution [release page](https://github.com/kubeflow/community-distribution/releases) or head over to the Getting Started and Support pages.

## Join the Community

We would like to thank everyone who contributed to Kubeflow 26.03, and especially Tarek Abouzeid for his work as the v26.03 Release Manager. We also extend our thanks to the entire release team and the working group leads, who continuously and generously dedicate their time and expertise to Kubeflow.

> Release team members : Tarek Abouzeid, Anya Kramar, Andy Stoneberg, Alyssa Goins, Matteo Mortari, Adysen Rothman, Jon Burdo, Milos Grubjesic, Vraj Bhatt, Dhanisha Phadate, Alok Dangre

Reviewer 7: Some names are missing from the release team. 

https://www.kubeflow.org/docs/kubeflow-distribution/releases/kubeflow-26.03/#kubeflow-community-distribution-2603

Reviewer 2: yes we shuld remove them. you can make direct code suggestions in the PR such that we only have to click on merge.

Working Group leads : Andrey Velichkevich, Julius von Kohout, Mathew Wicks, Matteo Mortari

> Kubeflow Steering Committee : Andrey Velichkevich, Julius von Kohout, Yuan Tang, Johnu George, Francisco Javier Araceo

Reviewer 7: some names don't match/ are missing, and some names are emeritus members, should they be listed?

https://github.com/kubeflow/community/tree/master/committee-steering#members

Reviewer 2: yes we shuld remove them. you can make direct code suggestions in the PR such that we only have to click on merge.

We would like to thank everyone who contributed to Kubeflow 26.03, and especially Tarek Abouzeid for his work as the v26.03 Release Manager. We also extend our thanks to the entire release team and the working group leads, who continuously and generously dedicate their time and expertise to Kubeflow.

Release team members : Tarek Abouzeid, Dominik Kawka, Dhanisha Phadate, Milind Dethe, Alok Dangre, Anya Kramar, Andy Stoneberg, Alyssa Goins, Matteo Mortari, Adysen Rothman, Milos Grubjesic, Vraj Bhatt

Working Group leads : Andrey Velichkevich, Julius von Kohout, Chase Christensen, Mathew Wicks, Francisco Arceo

> Kubeflow Steering Committee : Andrey Velichkevich, Julius von Kohout, Yuan Tang, Johnu George, Francisco Javier Araceo

Reviewer 7: ```suggestion
Kubeflow Steering Committee : Andrey Velichkevich, Chase Christensen, Francisco Arceo, Julius von Kohout, Mathew Wicks
```

Taken from: https://github.com/kubeflow/community/tree/master/committee-steering
